File: dotnet/Services/BeerCatalog.cs
using BeerVote.Models;

namespace BeerVote.Services;

public sealed class BeerCatalog
{
    private static readonly IReadOnlyList<Beer> Beers =
    [
        new(1, "Pilsner Urquell"),
        new(2, "Guinness Draught"),
        new(3, "Heineken"),
        new(4, "Budweiser Budvar"),
        new(5, "Hoegaarden"),
        new(6, "Paulaner Hefe-Weißbier"),
        new(7, "Sierra Nevada Pale Ale"),
        new(8, "BrewDog Punk IPA"),
        new(9, "Chimay Blue"),
        new(10, "Baltic Porter (local)"),
    ];

    private static readonly IReadOnlyDictionary<int, Beer> ById =
        Beers.ToDictionary(b => b.Id);

    public IReadOnlyList<Beer> ListAll() => Beers;

    public Beer? FindById(int id) => ById.GetValueOrDefault(id);

    public bool Exists(int id) => ById.ContainsKey(id);
}

File: mvps/php/app/Services/BeerCatalog.php
<?php

namespace App\Services;

class BeerCatalog
{
    /** @var list<array{id: int, name: string}> */
    private array $beers;

    /** @var array<int, array{id: int, name: string}> */
    private array $byId;

    public function __construct()
    {
        /** @var list<array{id: int, name: string}> $beers */
        $beers = config('beers');
        usort($beers, fn (array $a, array $b): int => $a['id'] <=> $b['id']);

        $this->beers = $beers;
        $this->byId = [];
        foreach ($beers as $beer) {
            $this->byId[$beer['id']] = $beer;
        }
    }

    /**
     * @return list<array{id: int, name: string}>
     */
    public function all(): array
    {
        return $this->beers;
    }

    /**
     * @return array{id: int, name: string}|null
     */
    public function findById(int $id): ?array
    {
        return $this->byId[$id] ?? null;
    }

    public function exists(int $id): bool
    {
        return isset($this->byId[$id]);
    }
}
